river/Cursor.zig

```diff
@@ -378,9 +407,16 @@ fn handleButton(listener: ?*c.wl_listener, data: ?*c_void) callconv(.C) void {
         }
     }
 
+    // Get the output where the cursor is
+    const root = &self.seat.input_manager.server.root;
+    var ox = self.wlr_cursor.x;
+    var oy = self.wlr_cursor.y;
+    // TODO should we rellly return here? Maybe catch unreachable
+    const output = Output.outputAt(root, &ox, &oy) orelse return;
+
     var sx: f64 = undefined;
     var sy: f64 = undefined;
-    if (self.surfaceAt(self.wlr_cursor.x, self.wlr_cursor.y, &sx, &sy)) |wlr_surface| {
+    if (output.surfaceAt(ox, oy, &sx, &sy)) |wlr_surface| {
```

Comment:
The fact that this logic is now duplicated at every `surfaceAt()` call is code smell. I'd prefer to keep the previous api which selected the output internally to `surfaceAt()`.

I'm also not 100% convinced that moving these functions to `Output` is the right move as they are only ever used in `Cursor.zig` and this PR does not change that. If they need to be accessed outside of `Cursor` in the future, `Root` would be the obvious place for them IMO.

Reply:
Yeah, I can't really recall why I though moving the functions was a good idea.
Moved them back now.